### PR TITLE
[PM-12074] - Refactored `Index` to use `UserModel`

### PR DESCRIPTION
**Pull request by jrmccannon**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-12074

## 📔 Objective
Instead of using the entity model for the User which is directly tied to the database, I switched out the model for a lightweight view model that can be used to hold only what that page needs.  This will allow it to be expanded for properties that aren't directly on the database model.  This also removes the feature flag code in the controller and view.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Admin/Controllers/UsersController.cs [hunk 1]
```diff
@@ -63,20 +56,11 @@ public async Task<IActionResult> Index(string email, int page = 1, int count = 2
 
         var skip = (page - 1) * count;
         var users = await _userRepository.SearchAsync(email, skip, count);
-
-        if (_featureService.IsEnabled(FeatureFlagKeys.MembersTwoFAQueryOptimization))
```
Comment (r-tome) [inline src/Admin/Controllers/UsersController.cs:67]: The feature flag lookup, should be kept here. We only want to execute that query if its enabled

File: src/Admin/Models/UserModel.cs
Comment (r-tome) [inline src/Admin/Models/UserModel.cs:1]: Instead of creating a new model can we refactor `UserViewModel`?

File: test/Admin.Test/Models/UserModelTests.cs
Comment (r-tome) [inline test/Admin.Test/Models/UserModelTests.cs:1]: Use `[SutProviderCustomize]` to automatically create the fixture and `[BitAutoData]` to create test data

**Review by r-tome — CHANGES_REQUESTED** [review]

File: src/Admin/Models/UserModel.cs
Comment (jrmccannon) [reply src/Admin/Models/UserModel.cs:1]: Yep! Also had to tweak edit model and all the `.cshtml` files to reflect this.

**Review by r-tome — APPROVED** [review]
Looks good to me! Thanks for making the changes I had requested.

Ensure that QA tests the editing of all User fields that should be modifiable

**Pull request merged**